Source/AwsCommonRuntimeKit/auth/signing/SigV4HttpRequestSigner.swift

```diff
@@ -31,58 +28,81 @@ public class SigV4HttpRequestSigner {
     ///    - `config`: The `SigningConfig` to use when signing.
     /// - `Throws`: An error of type `AwsCommonRuntimeError` which will pull last error found in the CRT
     /// - `Returns`: Returns a signed http request `HttpRequest`
-    public func signRequest(request: HttpRequest, config: SigningConfig) async throws -> HttpRequest {
-        typealias SignedContinuation = CheckedContinuation<HttpRequest, Error>
+    public static func signRequest(request: HttpRequest, config: SigningConfig, allocator: Allocator = defaultAllocator) async throws {
 
-        //TODO: Fix. This is weird. We should not throw make from last error here.
-        if config.configType != .aws || (config.credentialsProvider == nil && config.credentials == nil) {
-            throw CommonRunTimeError.crtError(.makeFromLastError())
+        guard let signable = aws_signable_new_http_request(allocator.rawValue, request.rawValue) else {
+            fatalError("Unable to create a new singable http request")
+        }
```

Comment:
This can only fail if we fail to init a list due to size. We can either do fatalError here or define a new error to throw.